### PR TITLE
APIGOV-21522 - Lookup for subscription and consumer orgID

### PR DESCRIPTION
**Pull request by vivekschauhan**
- Changes to use references sub resource in AccessRequest to lookup associated subscription 
- Changes to use marketplace sub resource in ManagedApplication to lookup consumer org identifier

### CODE REVIEW

**Review by sbolosan — DISMISSED** [review]

**Review by jcollins-axway — DISMISSED** [review]

File: pkg/agent/discoverycache.go [hunk 1]
```diff
@@ -344,7 +344,7 @@ func (j *discoveryCache) updateManagedApplicationCache() {
 	// TODO - Remove custom subresource and include subject subresource when added to model
```
Comment (jcollins-axway) [inline pkg/agent/discoverycache.go:344]: These comments still needed?

File: pkg/agent/handler/traceaccessrequest_test.go [hunk 2]
```diff
@@ -94,9 +95,9 @@ func TestTraceAccessRequestTraceHandler(t *testing.T) {
 	c.getRI = &v1.ResourceInstance{
 		ResourceMeta: v1.ResourceMeta{
 			Metadata: v1.Metadata{
-				ID: "subscription",
+				ID: "subscription-id",
```
Comment (jcollins-axway) [inline pkg/agent/handler/traceaccessrequest_test.go:98]: Should these be constants?

**Review by jcollins-axway — APPROVED** [review]

File: pkg/agent/handler/traceaccessrequest_test.go (hunk 2 above)
Comment (vivekschauhan) [reply pkg/agent/handler/traceaccessrequest_test.go:98]: Its test data, but have other string that I can create constants for. Will do the change

File: pkg/agent/discoverycache.go (hunk 1 above)
Comment (vivekschauhan) [reply pkg/agent/discoverycache.go:344]: Will remove it

**Review by sbolosan — APPROVED** [review]

**Pull request merged**